docs/3.0.0-beta.x/migration-guide/migration-guide-beta.19-to-beta.20.md

```diff
@@ -0,0 +1,221 @@
+        models[model.globalId] = formatModel(model);
+      });
+    });
+
+    Object.values(strapi.components).forEach(model => {
+      models[model.globalId] = formatModel(model);
+    });
+
+    fs.writeFileSync('models.json', JSON.stringify(models, null, 2));
+    process.exit(0);
+  });
+
+function formatModel(model) {
+  return {
+    collection: model.collectionName,
+    files: Object.keys(model.attributes).reduce((acc, key) => {
+      const attr = model.attributes[key];
+      if (attr.model === 'file' && attr.plugin === 'upload') {
+        acc[key] = 'single';
+      }
+
+      if (attr.collection === 'file' && attr.plugin === 'upload') {
+        acc[key] = 'multiple';
+      }
+      return acc;
+    }, {}),
+  };
+}
+```
+
+then run the script from the root of your project
+
+```sh
+node export.js
+```
+
+This script will create a models.json file at the root of your project that looks like:
+
+```json
+{
+  "ModelName": {
+    "collection": "collectionName",
+    "files": {
+      "image": "single",
+      "images": "multiple"
+    }
+  }
+}
+```
+
+### Create migration script
+
+You then need to create the mongo shell script to run to migrate your data strucutre. First create a `migration.js` file with the following code:
+
+```js
+var models = {
+  /* paste the object model.json  here */
+};
+
+for (var i in models) {
+  var model = models[i];
+  var update = {};
+  var keyCount = 0;
+
+  for (var key in model.files) {
+    keyCount += 1;
+    update[key] = '';
+  }
+
+  if (keyCount > 0) {
+    db.getCollection(model.collection).update({}, { $unset: update }, { multi: true });
+  }
+}
+
+var fileCursor = db.getCollection('upload_file').find({});
+
+while (fileCursor.hasNext()) {
+  var el = fileCursor.next();
+  el.related.forEach(function(fileRef) {
+    var model = models[fileRef.kind];
+
+    if (!model) {
+      return;
+    }
+
+    var fieldType = model.files && model.files[fileRef.field];
+
+    // stop if the file points to a field the user didn't specify
+    if (!fieldType) {
+      return;
+    }
+
+    if (fieldType === 'single') {
+      db.getCollection(model.collection).updateOne(
+        { _id: fileRef.ref },
+        { $set: { [fileRef.field]: el._id } }
+      );
+    } else if (fieldType === 'multiple') {
+      db.getCollection(model.collection).updateOne(
+        { _id: fileRef.ref },
+        { $push: { [fileRef.field]: el._id } }
+      );
+    }
+  });
+}
+```
+
+Then you will need to copy the content of `models.json` to the `models` variable at the top of this script:
+
+```js
+var models = {
+  ModelName: {
+    collection: 'collectionName',
+    files: {
+      image: 'single',
+      images: 'multiple',
+    },
+  },
+};
+
+// rest of the script
+```
+
+Finally you can load this script in your mongo shell and run it.
+
+## Rebuilding your administration panel
+
+Now delete the `.cache` and `build` folders. Then run `yarn develop`.
```

Comment:
What do you think about create a `clean` npm script?

```
"clean": "rm -rf .cache/ build/"
```

So we can use `yarn clean`.

Reply:
That's what I have (more or less) in my own project but it won't work on windows, you would need the rimraf package

Reply:
We would rather add --clean option to yarn build so it clears those folders directly

Reply:
I also have it in my projects, so I thought it would be good to add it to Strapi. 😁

Ok! So:

```
"clean": "rimraf -rf .cache/ build/"
```

Reply:
@alexandrebodin I like it!

Reply:
@alexandrebodin I agree it would make the process more seemless if the build command just deleted the .cache and build folders directly. (Honestly the .cache should be removed after the build is completed anyway)